eth/block_number_or_tag.go

```diff
@@ -9,9 +9,19 @@ import (
 type Tag string
 
 const (
-	TagLatest   Tag = "latest"
+	// TagLatest aka head block
+	TagLatest Tag = "latest"
+	// TagEarliest aka genesis
 	TagEarliest Tag = "earliest"
-	TagPending  Tag = "pending"
+	// TagSafe lags behind "unsafe" by around 4 seconds and is less likely to reorg
+	TagSafe Tag = "safe"
+	// TagUnsafe will be used as an alias to `TagLatest` to avoid backwards incompatibility.
+	TagUnsafe Tag = "unsafe"
+	// TagFinalized refers to a block that lags by one or two epochs (so 64-128 blocks)
+	// being "unsafe" and ~~will never reorg~~
```

Comment:
It will _typically_ lag by one or two epochs but can lag further during consensus issues.  I took a stab at a less definitive but more technically correct description below:

```suggestion
	// TagFinalized refers to a block that typically lags by one or two epochs (so 64-128 blocks)
	// but can lag further during consensus issues.  Once finalized a block can only reorg via a hard fork.
```